File: go/etl/etl.go
// Package etl extracts transforms and loads data
package etl

import "strings"

// Transform transforms a map of point -> characters into a map of character -> point
// for a game of scrabble.
func Transform(data map[int][]string) map[string]int {
	transformed := map[string]int{}

	for key, value := range data {
		for _, char := range value {
			transformed[strings.ToLower(char)] = key
		}
	}

	return transformed
}

File: cpp/etl/etl.cpp
#include "etl.h"

namespace etl {

	std::map<char, int> transform(std::map<int, std::vector<char>> old)
	{
		auto key_iter = old.begin();
		std::map<char, int> transformed;

		for (; key_iter != old.end(); key_iter++)
		{
			auto value_iter = key_iter->second.begin();
			for(; value_iter != key_iter->second.end(); value_iter++)
				transformed[*value_iter + 32] = key_iter->first;
		}

		return transformed;
	}

}  // namespace etl
